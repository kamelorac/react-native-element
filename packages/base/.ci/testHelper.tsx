import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
