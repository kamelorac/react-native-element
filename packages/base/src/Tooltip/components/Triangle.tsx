import React from 'react';
import {
  View,
  StyleSheet,
  StyleProp,
  ViewStyle,
  ColorValue,
} from 'react-native';

type TriangleProps = {
  pointerColor?: ColorValue;
  style?: StyleProp<ViewStyle>;
  isDown?: boolean;
};

const Triangle: React.FC<TriangleProps> = ({ style, pointerColor, isDown }) => (
  <View
    testID="RNE__Tooltip_Triangle"
    style={StyleSheet.flatten([
      styles.triangle,
      {
        borderBottomColor: pointerColor,
      },
      style,
      isDown ? styles.down : {},
    ])}
  />
);

const styles = StyleSheet.create({
  down: {
    transform: [{ rotate: '180deg' }],
  },
  triangle: {
    width: 0,
    height: 0,
    backgroundColor: 'transparent',
    borderStyle: 'solid',
    borderLeftWidth: 8,
    borderRightWidth: 8,
    borderBottomWidth: 15,
    borderLeftColor: 'transparent',
    borderRightColor: 'transparent',
    borderBottomColor: 'white',
  },
});

export default React.memo(Triangle);
