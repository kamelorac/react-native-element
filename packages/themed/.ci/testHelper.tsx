import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
