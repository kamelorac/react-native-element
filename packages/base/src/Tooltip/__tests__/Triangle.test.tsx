import React from 'react';
import Triangle from '../components/Triangle';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('Tooltip component', () => {
  it('should match snapshot ', () => {
    const { toJSON } = renderWithWrapper(<Triangle />, 'RNE__Tooltip_Triangle');
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render ', () => {
    const { wrapper } = renderWithWrapper(
      <Triangle pointerColor="red" />,
      'RNE__Tooltip_Triangle'
    );
    expect(wrapper.props.style.borderBottomColor).toBe('red');
  });

  it('should render without downward', () => {
    const { wrapper } = renderWithWrapper(
      <Triangle isDown />,
      'RNE__Tooltip_Triangle'
    );
    expect(wrapper.props.style.transform).toMatchObject([{ rotate: '180deg' }]);
  });
});
