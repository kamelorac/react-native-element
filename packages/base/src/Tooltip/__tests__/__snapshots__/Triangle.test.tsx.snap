// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component should match snapshot  1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "borderBottomColor": undefined,
      "borderBottomWidth": 15,
      "borderLeftColor": "transparent",
      "borderLeftWidth": 8,
      "borderRightColor": "transparent",
      "borderRightWidth": 8,
      "borderStyle": "solid",
      "height": 0,
      "width": 0,
    }
  }
  testID="RNE__Tooltip_Triangle"
/>
`;
